keps/sig-cli/2551-return-code-normalization/README.md

```diff
@@ -0,0 +1,652 @@
+wants to warn users when the apply command fails because of differences between the manifests.
+
+#### Story 2
+Bruce Wayne, the security administrator of the Gotham Inc Company is following the development of a new product. Bruce asked
+for the developers to warn when a new deployment fails because of the lack of some permission, so those permissions can be 
+updated for the pipeline to work correctly.
+
+The developers are making a lot of changes, and they keeps asking for Bruce to look for every pipeline execution, even those that 
+fails because of wrong manifests and not because of authorization issues. So Bruce needs a new mechanism: That the pipeline knows
+when it fails because of the lack of the authorization on Kubernetes API Server, and then the warning is sent to the security team
+only when the pipeline breaks because of a specific error code that represents this authorization failure.
+
+#### Story 3
+Roberta works as the product manager of a big CI/CD SaaS provider. They want to have in their marketplace the
+execution of kubernetes commands targeting a cluster, and fastly providing a feedback to the user if the error
+was due to something on the client side (like a missing flag, an invalid yaml file because of...tabs...) or if this
+is due to some invalid operation on the server side.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+* Users already relying on specific error codes might face migration problems / false positives or negatives 
+because of some specific error code change. The proposal here is to decide if the old or new return code behavior
+will be used based on a well defined environment variable. The old behavior will be kept and followed by the 
+deprecation path will be defaulted after a number of releases.
+
+```
+<<[UNRESOLVED feature gating ]>>
+Decide if a global flag to control this behavior will be present, or if the Env Variable is enough
+<<[/UNRESOLVED]>>
+```
+
+
+* Some exit code might not be proper related to an error. The reserved exit codes are documented [here](https://tldp.org/LDP/abs/html/exitcodes.html)
+and should be used carefully in a way to not generate conflict with existing scripts. 
+
+```
+<<[UNRESOLVED initial error codes ]>>
+Define the initial error codes, and how to document them
+<<[/UNRESOLVED]>>
+```
+
+## Design Details
+
+The majority of commands already are organized as the following:
+* Run Complete to complete missing information with defaults. This runs on the client side
+* Run Validation to check command syntax and missing arguments. This runs on the client side
+* Run the command itself. This might run on the client side, be dry-run, run an external command or call the APIServer.
```

Comment:
This might actually change during the separattion of cobra from `Options` struct, so it would be nice to put it in a more generic way.

Reply:
@deejross will take a look on this